### ISSUE TITLE
LibJS: Unable to use keywords as property names in object literals

### ISSUE THREAD

**Issue by Lubrsi** [SerenityOS/serenity#11078]
For example: `{ url: "", async: false }` produces:
```
{ url: "", async: false }
                ^
Uncaught exception: [SyntaxError] Unexpected token Colon. Expected Semicolon (line: 1, column: 17)
```

Similarly, `{ url: "", function: false }` produces:
```
{ url: "", function: false }
                   ^
Uncaught exception: [SyntaxError] Unexpected token Colon. Expected ParenOpen (line: 1, column: 20)
```

This affects jQuery for example: https://github.com/jquery/jquery/blob/a684e6ba836f7c553968d7d026ed7941e1a612d8/src/ajax.js#L304

**Comment by davidot** [comment]
What is the exact code you use to reproduce this? 
Because `{ url: "", function: false }` isn't valid as it is a block with label url then a comma expression with an invalid function.
And the same for the original example `{url: "", async: false}`.
However for `async` there are broken things like:
```js
const obj = {
    async: "",
}
```
But as far as I can see this work fine with `function`

(I already have a fix but want to bundle it with a couple of other parser fixes)

**Comment by Lubrsi** [comment]
The exact code was from minified jQuery of the code I linked to. I just took `async` out and noticed it was in an object literal (neglecting it's being used as an argument) and then tested it with `function` to see if that also worked. I had added `url` as a dummy property as the keyword by itself parsed and returned the value I had set it to. I had only tested this in Chrome and Node and neglected to test it other engines, so it seems I accidentally found a parser differential in V8 as this doesn't work in SpiderMonkey or JSC either :^)
